plugins/BEdita/Core/src/Model/Action/ListEntitiesAction.php

```diff
@@ -104,11 +104,11 @@ protected function buildFilter(Query $query, array $filter)
             if ($this->Table->associations()->has($camelizedKey)) {
                 // Associated match (primary key only).
                 $target = $this->Table->association($camelizedKey)->getTarget();
-                $targetPrimaryKey = array_map(
-                    [$target, 'aliasField'],
-                    (array)$target->getPrimaryKey()
-                );
-                $conditions = array_combine($targetPrimaryKey, (array)$value);
+                $targetPrimaryKey = $target->aliasField($target->getPrimaryKey());
```

Comment:
:thinking:  I'm not sure that is the right solution. I think the previous code was done for supporting multiple primary keys.

Maybe we would build `$targetPrimaryKey` as previous and set the `IN` conditions just if  `count($targetPrimaryKey)` is equal to `1` otherwise I think the `array_combine` should be used.

@fquffio am I wrong?

Reply:
You are right. However, right now we don't use composite primary keys, except for some junction tables that are not "exposed" as standalone resources. Therefore, it should be (quite) safe.

Also, there is another problem here: the `INNER JOIN` strategy was definitely not the best choice in the first place: it makes SQL return multiple times the same row for "to many" associations, and forces us to use `DISTINCT` (see line 114). A better approach would likely involve `EXISTS` (i.e.: `SELECT <fields> FROM <table> WHERE EXISTS (<subquery>)`), but in this case we should build a subquery — and that subquery would be different according to the type of association.

Reply:
OK! I think we can live with it :smile_cat: